### DISCUSSION TITLE
Editing contributors pull requests denied locally and github.dev but not on github.com

### DISCUSSION THREAD

**Discussion by kaovilai** [community/community#34024] [Pull Requests]
Looking at https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally#modifying-an-active-pull-request-locally.

The documentation seems to suggests that maintainers/collaborators cannot push to remote PR refs.

However I have been able to edit one file at a time in web editor [at a file page](https://github.com/sseago/oadp-operator/edit/update-velero-images/controllers/restic_test.go) by clicking on 
![image](https://user-images.githubusercontent.com/11228024/192071718-0c02ef11-8431-4021-b9f5-9728b869df03.png)
and commit directly to the PR author's branch.

The following scenarios do not work however.
- github.dev online editor (where you use `.` shortcut to enter) tells you that you do not have write access to this branch
- git checkout and pushing to this remote is denied similarly to below [example](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally#error-failed-to-push-some-refs)

```
! [remote rejected] HEAD -> refs/pull/1/head (deny updating a hidden ref)
error: failed to push some refs to 'git@github.local:USERNAME/REPOSITORY.git'
```

So I would just like to get clarity on this.. if you can edit one file at a time online on github.com, why can't we do the same on github.dev or from a local (macos) git checkout (SSH)?

**Comment by airtower-luna** [comment]
The documentation explicitly [states that you cannot push to `refs/pull/**`](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally#error-failed-to-push-some-refs). You need to create your own branch and push that.

If you want to push directly to the PR branch (assuming you have permission) don't use the `refs/pull/**` ref, push directly to the right repository and branch instead.

**Reply by kaovilai** [reply]
I tried both push locations. Doesn't work. GitHub.dev doesn't work. Yet if I stay on GitHub.com and edit a single file on that branch it commits successfully..
